lib/openhab/core/emulate_hash.rb

```diff
@@ -0,0 +1,233 @@
+      end
+
+      # @!visibility private
+      def assoc(key)
+        to_hash.assoc(key.to_s)
+      end
+
+      # @!visibility private
+      def clear
+        replace({})
+      end
+
+      # @!visibility private
+      def merge!(*others, &block)
+        return self if others.empty?
+
+        new_config = to_h
+        others.each do |h|
+          new_config.merge!(h.transform_keys(&:to_s), &block)
+        end
+        replace(new_config)
+      end
+      alias_method :update, :merge!
+
+      # @!visibility private
+      def dig(*keys)
+        to_hash.dig(*keys.map(&:to_s))
+      end
+
+      # @!visibility private
+      def compact!
+        replace(compact)
+      end
+
+      # @!visibility private
+      def slice(*keys)
+        to_h.slice(*keys.map(&:to_s))
+      end
+
+      # @!visibility private
+      def delete_if(&block)
+        raise NotImplementedError unless block
+
+        replace(to_h.delete_if(block))
+      end
+
+      # @!visibility private
+      def keep_if(&block)
+        select!(&block)
+        self
+      end
+
+      # @!visibility private
+      def except(*keys)
+        to_hash.except(*keys.map(&:to_s))
+      end
+
+      # @!visibility private
+      def select!(&block)
+        raise NotImplementedError unless block
+
+        r = to_hash.select!(&block)
+        replace(r) if r
+      end
+      alias_method :filter!, :select!
+
+      # @!visibility private
+      def reject!(&block)
+        raise NotImplementedError unless block
+
+        r = to_hash.reject!(&block)
+        replace(r) if r
+      end
+
+      # @!visibility private
+      def fetch_values(*keys, &block)
+        to_hash.fetch_values(*keys.map(&:to_s), &block)
+      end
+
+      # @!visibility private
+      def transform_keys!(*args, &block)
+        replace(transform_keys(*args, &block))
+      end
+
+      # @!visibility private
+      def transform_values!(&block)
+        raise NotImplementedError unless block
+
+        replace(transform_values(&block))
+      end
+
+      # @!visibility private
+      def values_at(*keys)
+        to_hash.values_at(*keys.map(&:to_s))
+      end
+
+      # @!visibility private
+      def <(other)
+        to_hash < other.to_hash.transform_keys(&:to_s)
+      end
+
+      # @!visibility private
+      def <=(other)
+        to_hash <= other.to_hash.transform_keys(&:to_s)
+      end
+
+      # # @!visibility private
+      # def ==(other)
+      #   return to_hash == other.transform_keys(&:to_s) if other.is_a?(Hash)
+      #
+      #   equals(other)
+      # end
+
+      # @!visibility private
+      def >(other)
+        to_hash > other.to_hash.transform_keys(&:to_s)
+      end
+
+      # @!visibility private
+      def >=(other)
+        to_hash >= other.to_hash.transform_keys(&:to_s)
+      end
+
+      # @!visibility private
+      def compare_by_identity
+        raise NotImplementedError
+      end
+
+      # @!visibility private
+      def compare_by_identity?
```

Comment:
this should probably not be implemented explicitly here, since it's delegated to `to_hash` above (and can be delegated to `to_map` instead). NamespaceHash can keep its override, to a void unnecessarily creating a temporary object when delegating